modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQueryParser.java

```diff
@@ -1181,8 +1181,12 @@ private GridSqlCreateTable parseCreateTable(CreateTable createTbl) {
 
         LinkedHashMap<String, GridSqlColumn> cols = new LinkedHashMap<>(data.columns.size());
 
-        for (Column col : data.columns)
+        for (Column col : data.columns) {
+            if (cols.containsKey(col.getName()))
+                throw new IgniteSQLException("Duplicate column name: " + col.getName(), IgniteQueryErrorCode.PARSING);
```

Comment:
You could just analyze return value of `cols.put` instead of using `containsKey`

Reply:
Done.